server/services/referralService.ts

```diff
@@ -40,6 +40,18 @@ export default class ReferralService {
     return referralClient.findMyReferralSummaries(query)
   }
 
+  async getNumberOfTasksCompleted(username: Express.User['username'], referralId: Referral['id']): Promise<number> {
+    const referralProgressIndicatorKeys = [
+      'hasReviewedProgrammeHistory',
+      'oasysConfirmed',
+      'additionalInformation',
+    ] as const
+
+    const referral = await this.getReferral(username, referralId)
+
+    return 1 + referralProgressIndicatorKeys.filter(item => referral[item]).length
```

Comment:
Would checking for the presence of a `prisonNumber` on the Referral be a bit clearer here than using `1`? The commit message does explain this, but might be easier for future readers to understand.